### PR TITLE
chore: remove lazy loading relationships in list

### PR DESCRIPTION
**Pull request by DanRibbens**
## Description

Recently the collection List was changed to fetching the table data with a depth of 0 and instead allow relationship cells in the table to make calls for missing data through a new provider for efficiency. We decided to roll this change back in favor of having a depth of 1 so that one request for table data can populate any needed relationships for displaying in the table. It is a tradeoff between making lots of smaller queries vs 1 bigger query. With the API improvements made with the changes in data loader, it is likely better to use population on related data.

TLDR: Reverts some optimizations that were done in favor of using dataLoader for collection list.

Related issue: https://github.com/payloadcms/payload/issues/822

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] Existing test suite passes locally with my changes

### CODE REVIEW

**Comment by DanRibbens** [comment]
Based on some performance testing we decided to keep the optimization in and address the original issue.
